### ISSUE TITLE
Parallax Factor documentation not clear

### ISSUE THREAD

**Issue by Quillraven** [mapeditor/tiled#3196]
Hi,

I have a question because I am working on a PR for LibGDX to support the parallax factor rendering that was introduced with Tiled 1.5.
However, I do not really understand what exactly that factor means and also what is represented to me in the editor itself.

Take a look at following screenshot:
![image](https://user-images.githubusercontent.com/93260/144457370-e7a6e1b0-dcf3-47ab-af4f-fa35f1598309.png)

I have a x-factor of 1.25 and it seems like the layer gets moved to the left by ~1.4 tiles and gets then of course moved further when I scroll to the right/left? But where do I see my camera position? Is this left movement already happening for a camera position of 0,0?
Also, what is the exact formula of the rendering? Is the camera position simply multiplied with the factor values? Or are there other factors like viewport width/height or map width/height or tile width/height taken into account?

In the end I am trying to get the same render result with LibGDX as we have it in Tiled but to make that happen I need to better understand what exactly Tiled is doing. The documentation that I found is not really detailed :(

Maybe it is some well known approach and everybody knows what that factor means but I couldn't find anything so far, sorry.

**Comment by Quillraven** [comment]
Thank you for the explanation! I have still one question please.

You mention that it takes the distance between the origin and the center of the view. Which distance are you talking about because let's take the 2D-distance formula which is sqrt(diffX*diffX + diffY*diffY). When I scroll on the Y-axis the distance between the view and origin obviously changes. However, this seems to have no impact on the parallax scrolling on the x-axis.

Is it then maybe simply like this?
parallax x offset = diffX * (1f - scrolling factor x)
parallax y offset = diffY * (1f - scrolling factor y)

Where diffX/diffY is the x/y difference of origin and view.

**Comment by bjorn** [comment]
Well, let's just say the distance between the origin and the center of the view is given by a vector `vec(diffX, diffY)`, indeed it makes no sense to calculate the length of that vector since we're applying the parallax offset to in each dimension separately.

And then it should just be:
```
offsetX = diffX * parallaxX
offsetY = diffY * parallaxY
```
This is the offset the layer should have from the origin. So when a layer would normally have moved 10 pixels to the right, but it has a parallax factor of 0.5, it should instead be rendered only 5 pixels to the right.

But, it all gets a bit more complicated, since usually we're not moving layers around, but the camera, and this is causing the layers to move in the view. In this case, we're moving layers around to compensate and that is where your math comes in:
```
layerX = diffX * (parallaxX - 1)
layerY = diffY * (parallaxY - 1)
```
Now, when a layer has moved 10 pixels to the right, we're setting its position to -5 to make sure that it really only moves 5 pixels on the screen.

In Tiled's code, I take the position in the center of the view (in map coordinates). This position goes up when the view shifts left, which is why it's the other way around there:
```
layerX = centerX * (1 - parallaxX)
layerY = centerY * (1 - parallaxY)
```
When the position in the center is 10, the layers will have normally moved 10 pixels to the left (-10), but by positioning the layer at 5, I'm making sure that it only moves 5 pixels to the left.

I'm really wondering now what's the easiest way to document this, heh.

**Comment by Quillraven** [comment]
Cool, thank you for the detailed explanation! Is there actually a way to see the current view coordinates? I think it could be useful to see those numbers to better understand how the rendering is performed.

Anyway, thank you again 👍🙂

**Comment by bjorn** [comment]
> Cool, thank you for the detailed explanation! Is there actually a way to see the current view coordinates? I think it could be useful to see those numbers to better understand how the rendering is performed.

At the moment you can only see the position below the mouse, in the status bar. If you position the mouse in the center of the view, it would roughly tell you the value used to calculate the parallax offset.

Btw, to get a better match between Tiled preview and the game, it is planned to implement a "viewport", with a defined size, and potentially what to use as the reference point within that viewport (so it could be top-left instead of center, for example). This is covered by issue #2962.

**Comment by bjorn** [comment]
I've pushed an addition to the manual regarding the reference points (will be available at https://doc.mapeditor.org/en/latest/manual/layers.html#parallax-scrolling-factor soon).

Please let me know whether this part helps or whether you see any issue or remaining need for clarification!

**Comment by Quillraven** [comment]
Looks good to me! I also made the PR on LibGDX site a few days ago: [LibGDX PR](https://github.com/libgdx/libgdx/pull/6714)

I think I was able to reproduce the result of Tiled correctly. The only difference is that my PR is loading the scrolling factor correctly for hierarchical layers. I think you mentioned that in Tiled there is currently a bug and it is not represented according to the documentation, correct?
